62_CAD/main.cpp

```diff
@@ -900,25 +900,53 @@ class ComputerAidedDesign final : public examples::SimpleWindowedApplication, pu
 		
 		// Loading font stuff
 		m_textRenderer = nbl::core::make_smart_refctd_ptr<TextRenderer>();
-		m_arialFont = nbl::core::make_smart_refctd_ptr<FontFace>(core::smart_refctd_ptr(m_textRenderer), std::string("C:\\Windows\\Fonts\\arial.ttf"));
 
-		const auto str = "MSDF: ABCDEFGHIJKLMNOPQRSTUVWXYZ abcdefghijklmnoprstuvwxyz '1234567890-=\"!@#$%¨&*()_+";
+		m_arialFont = nbl::core::make_smart_refctd_ptr<FontFace>(core::smart_refctd_ptr(m_textRenderer), std::string("C:\\Windows\\Fonts\\arial.ttf"));
+		const auto str = "MSDF: ABCDEFGHIJKLMNOPQRSTUVWXYZ abcdefghijklmnoprstuvwxyz '1234567890-=\"!@#$%&*()_+";
 		singleLineText = std::unique_ptr<SingleLineText>(new SingleLineText(
 			core::smart_refctd_ptr<FontFace>(m_arialFont), 
 			std::string(str)));
 
-
 		drawResourcesFiller.setGlyphMSDFTextureFunction(
-			[&](nbl::ext::TextRendering::FontFace* face, uint32_t glyphIdx) -> core::smart_refctd_ptr<asset::ICPUBuffer>
+			[&](nbl::ext::TextRendering::FontFace* face, uint32_t glyphIdx) -> core::smart_refctd_ptr<asset::ICPUImage>
 			{
-				return face->generateGlyphMSDF(MSDFPixelRange, glyphIdx, drawResourcesFiller.getMSDFResolution());
+				return face->generateGlyphMSDF(MSDFPixelRange, glyphIdx, drawResourcesFiller.getMSDFResolution(), 4);
 			}
 		);
 
 		drawResourcesFiller.setHatchFillMSDFTextureFunction(
-			[&](HatchFillPattern pattern) -> core::smart_refctd_ptr<asset::ICPUBuffer>
+			[&](HatchFillPattern pattern) -> core::smart_refctd_ptr<asset::ICPUImage>
 			{
-				return Hatch::generateHatchFillPatternMSDF(m_textRenderer.get(), pattern, drawResourcesFiller.getMSDFResolution());
+				ICPUImage::SCreationParams imgParams;
+				{
+					imgParams.flags = static_cast<ICPUImage::E_CREATE_FLAGS>(0u); // no flags
+					imgParams.type = ICPUImage::ET_2D;
+					imgParams.format = TextRenderer::MSDFTextureFormat;
+					imgParams.extent = { uint32_t(MSDFSize), uint32_t(MSDFSize), 1 };
+					imgParams.mipLevels = 1u;
+					imgParams.arrayLayers = 1u;
+					imgParams.samples = ICPUImage::ESCF_1_BIT;
+				}
+
+				auto buffer = Hatch::generateHatchFillPatternMSDF(m_textRenderer.get(), pattern, drawResourcesFiller.getMSDFResolution());
```

Comment:
Hatch::generateHatchFillPatternMSDF
you know this function should also return `IGPUImage` + the image should have the same mip levels as expected in our texture array, it's true these fill patterns are fixed sized on screen, but we don't want to make the assumption at this level to be able to have flexibility in the future to change those.